code/modules/projectiles/guns/projectile.dm

```diff
@@ -24,6 +24,7 @@
 	//For MAGAZINE guns
 	var/magazine_type = null	//the type of magazine that the gun comes preloaded with
 	var/obj/item/ammo_magazine/ammo_magazine = null //stored magazine
+	var/allowed_magazines		//determines list of which magazines will fit in the gun

```

Comment:
Since this is and should be a list, change the declaration to:
`var/list/allowed_magazines`

There's no real reason not to.